### PR TITLE
Fix #999 and add tests

### LINKED ISSUE #999: Resize (ResizeMode.Max) only with height restraint producing image 1px too tall?
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [x] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
I'm resizing images of varying dimensions to a maximum height of 450px (any width), often the image produced is actually 451px high. 

### Steps to Reproduce

As an example, with this image: https://sample-videos.com/img/Sample-jpg-image-5mb.jpg

~~~ csharp
using (Image image = Image.Load(imgFile))
{
    image.Mutate(x =>
        x.Resize(
            new ResizeOptions()
            {
                Mode = ResizeMode.Max,
                Size = new Size() { Height = 450 }
            }
        )
    );
    image.Save(outImgFile2)
}
~~~

This produces an image from 5072x6761px -> 338x451px exceeding the maximum 450 height specified.

As a work around, If I also specify a width, i.e. 

~~~ csharp
using (Image image = Image.Load(imgFile))
{
    image.Mutate(x =>
        x.Resize(
            new ResizeOptions()
            {
                Mode = ResizeMode.Max,
                Size = new Size(int.MaxValue, 450)
            }
        )
    );
    image.Save(outImgFile2)
}
~~~

It produces an image of 338x450px as expected. 

Is the first example the intended behaviour or is it a bug? 

### System Configuration

- ImageSharp version: SixLabors.ImageSharp 1.0.0-beta0007
- Other ImageSharp packages and versions: n/a
- Environment (Operating system, version and so on): Windows 10 
- .NET Framework version: .NET Core 2.2

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Fixes #999 and adds addition `ResizeHelper` tests

### CODE REVIEW

File: src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor.cs [hunk 1]
```diff
@@ -26,28 +26,28 @@ public ResizeProcessor(IResampler sampler, int width, int height, Size sourceSiz
         {
             Guard.NotNull(sampler, nameof(sampler));
 
-            // Ensure size is populated across both dimensions.
+            // Ensure target size is populated across both dimensions.
             // If only one of the incoming dimensions is 0, it will be modified here to maintain aspect ratio.
             // If it is not possible to keep aspect ratio, make sure at least the minimum is is kept.
-            const int min = 1;
+            const int Min = 1;
             if (width == 0 && height > 0)
```
Comment (antonfirsov) [inline src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor.cs]: I wonder if we share code with the other method.

Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessor.cs]: I've given the code another once over to remove duplication and respond to comments. Much cleaner now!

File: src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeHelper.cs [hunk 2]
```diff
@@ -30,17 +28,32 @@ internal static class ResizeHelper
...
+            {
+                width = (int)MathF.Max(Min, MathF.Round(sourceSize.Width * height / (float)sourceSize.Height));
+            }
+
+            if (height == 0 && width > 0)
+            {
+                height = (int)MathF.Max(Min, MathF.Round(sourceSize.Height * width / (float)sourceSize.Width));
+            }
+
+            Guard.MustBeGreaterThan(width, 0, nameof(width));
```
[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeHelper.cs]: Agreed

File: src/ImageSharp/Processing/ResizeMode.cs [hunk 3]
```diff
@@ -42,6 +42,11 @@ public enum ResizeMode
         /// <summary>
         /// Stretches the resized image to fit the bounds of its container.
         /// </summary>
-        Stretch
+        Stretch,
+
+        /// <summary>
+        /// The target location and size of the resized image has been manually set.
+        /// </summary>
+        Manual
```
Comment (antonfirsov) [inline src/ImageSharp/Processing/ResizeMode.cs]: Wow, nice solution! 👍
(I was stuck to find a clean way to manage this.)

Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/ResizeMode.cs]: Yeah, surprised myself there!

**Pull request merged**